build.gradle

```diff
@@ -73,6 +82,22 @@ subprojects {
   apply plugin: 'nebula.dependency-recommender'
   apply plugin: 'java-library'
 
+  // apply this plugin in a try-catch block so that we can handle cases without .git directory
+  // apply to subprojects so that git properties information is available for runtime debugging and tests
+  //  e.g. sending build SHA as headers
+  try {
+    apply plugin: 'com.gorylenko.gradle-git-properties'
+    gitProperties {
+      gitPropertiesName = "${project.name}-version.properties"
+      extProperty = 'gitProps'
+      failOnNoGitDirectory = false
+    }
```

Comment:
This generates a file that is included in the runtime classpath of each subproject (or alternatively we could just do `iceberg-core`).

This would allow users to `./gradlew build` from the repository and get a Spark / Flink bundle that already has the git attributes file in it, as it would be included from `file("${rootDir}/core/build/resources/main/iceberg-core-version.properties)")` and then be included in the bundle jars etc automatically (provided the `.git` repository is still present).

The file naming would need to be reconciled, but that's the reasoning for having both. Happy to reduce it down, but just the one `build/iceberg-build.properties` file  generated in the block starting at line 45 wouldn't be added to JARs / available on the classpath in many situations.